File: crates/rr-core/tests/fixtures/ruby/surface.rb
# Documented service.
class Service < Base
  ATTEMPTS = 3

  def initialize(name)
    @name = name
  end

  def run
    helper(@name)
  end

  def _hidden
    1
  end

  private

  def secret
    2
  end
end

module Util
  def self.build
    Service.new('x')
  end
end

File: crates/rr-core/tests/fixtures/javascript/surface.js
import { helper } from './helper.js';

/** Documented class. */
export class Service extends Base {
  #secret = 1;
  constructor(name) {
    super();
    this.name = name;
  }
  run() {
    return helper(this.name);
  }
  static build() {
    return new Service('x');
  }
  get size() {
    return 1;
  }
}

export function bare(a) {
  return a;
}

export const arrow = (a) => a;
const value = 3;
export default Service;
